### ISSUE TITLE
WARN: unable to connect to default domain. Computer is likely not attached to an AD domain”

### ISSUE THREAD

**Issue by zakird** [zakird/pyad#31]
In some situations, users still receive an error when a root domain is not available because the machine is not joined to a domain. You should be able to import pyad, call pyad.set_defaults(ldap_server= "a.b.c.d", username="xyz\compsvc", password="xyzabc") and move on.

**Comment by Walleye007** [comment]
I have corrected the issue with connection to an AD server in adquery.py. The problem lies in the constant that was used in the code. It should have been Hex 201 not decimal 201. I have attached the modified file that corrects this issue. 

I also added in a call to "_set_defaults(options)". Now you can pass the default username and password into the query properly.

Below is a diff from pyad-0.5.14.py2.7.eggs\adquery.py Let me know if you would like the source code. 

  adquery.py (03 Jul 2015 15:13:26)
           ..\adquery.py (28 Sep 2015 15:43:32)
===================
       6       6  |    # Active Directory will use Kerberos, and possibly NTLM,
       7       7  |    # to authenticate the client.
+              8  |    #ADS_SECURE_AUTHENTICATION = 1
-      8          |    ADS_SECURE_AUTHENTICATION = 1
       9       9  |    #ADS_SECURE_AUTHENTICATION = 201 #server authentication
+             10  |    ADS_SECURE_AUTHENTICATION = 0x201 #server authentication
      10      11  |    # Requires ADSI to use encryption for data
      11      12  |    # exchange over the network.
===================
      28      29  |    
      29      30  |    def __init__(self, options={}):
+             31  |        self._set_defaults(options)
      30      32  |        self.__adodb_conn = win32com.client.Dispatch("ADODB.Connection")
      31      33  |        if self.default_username and self.default_password:
===================
      34      36  |            self.__adodb_conn.Properties("Password").Value = self.default_password
      35      37  |            self.__adodb_conn.Properties("Encrypt Password").Value = True
+             38  |            #adsi_flag = ADQuery.ADS_SECURE_AUTHENTICATION | \
+             39  |            #                ADQuery.ADS_USE_ENCRYPTION
+             40  |            adsi_flag = ADQuery.ADS_SECURE_AUTHENTICATION
-     36          |            adsi_flag = ADQuery.ADS_SECURE_AUTHENTICATION | \
-     37          |                            ADQuery.ADS_USE_ENCRYPTION
      38      41  |            self.__adodb_conn.Properties("ADSI Flag").Value = adsi_flag
-     39          |            print "adquery.py: using userid and password adsi_flag: ", adsi_flag
      40      42  |            self.__adodb_conn.Open("Active Directory Provider")
      41      43  |        else:
===================

**Comment by zakird** [comment]
I don't think that diff came through as intended. Can you email me the file you changed?

**Comment by Walleye007** [comment]
You are right! Will send it to you.

**Comment by Walleye007** [comment]
Zakir,

Here you go!

Cheers,

Pete.

**Comment by zakird** [comment]
I think that went to Github (which swallowed the attachement) not me

**Comment by Edlopushansky** [comment]
Hey , guys ,
Is there any resolution to this issue. ?
